### PR TITLE
Add new test file to validate customer details

### PR DESCRIPTION
**Pull request by singlefleck**

### CODE REVIEW

File: test/customer_test.exs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  test "invalid by name", state do
+    valid_customer = state[:valid_customer]
+
+    invalid_customer_by_name = %{valid_customer | name: "ra"}
+    refute Customer.is_valid(invalid_customer_by_name)
+
+    invalid_customer_by_name = %{valid_customer | name: "1bor"}
+    refute Customer.is_valid(invalid_customer_by_name)
+
+
```
Comment (blackode) [inline test/customer_test.exs]: remove one blank line

File: test/customer_test.exs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  use ExUnit.Case
+  doctest Customer
+
+  setup_all do
+    valid_customer = %{name: "boy", aadhar: "561664075050"}
+    invalid_customer= %{}
+
+    {:ok, valid_customer: valid_customer, invalid_customer: invalid_customer}
+
+
```
Comment (blackode) [inline test/customer_test.exs]: remove one blank line

File: test/customer_test.exs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+defmodule CustomerTest do
```
Comment (blackode) [inline test/customer_test.exs]: Update module name as `Registration.CustomerTest`

File: test/customer_test.exs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      invalid_customer_by_aadhar = %{valid_customer | aadhar: "1234"}
+      refute Customer.is_valid(invalid_customer_by_aadhar)
+
+
+      invalid_customer_by_aadhar = %{valid_customer | aadhar: "12345678901234"}
+      refute Customer.is_valid(invalid_customer_by_aadhar)
+
+    end
+
+
```
Comment (blackode) [inline test/customer_test.exs]: remove blank line

**Review by blackode — CHANGES_REQUESTED** [review]

File: lib/registration/customer.ex [hunk 5]
```diff
@@ -0,0 +1,40 @@
+defmodule Registration.Customer do
+         def is_valid(customer_details) do
```
Comment (blackode) [inline lib/registration/customer.ex]: define type and spec and doc

File: lib/registration/customer.ex [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+              %{name: name, aadhar: aadhar} = customer_details
+
+                if  check_name(name) && check_aadhar(aadhar) == true do
+                  true
+                end
```
Comment (blackode) [inline lib/registration/customer.ex]:
```suggestion
                check_name(name) && check_aadhar(aadhar)
               ```

File: lib/registration/customer.ex [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+         end
+
+        def check_name(name) do
+          if String.valid?(name) == true do
+            if String.length(name) >= 3 do
+              IO.inspect(name)
```
Comment (blackode) [inline lib/registration/customer.ex]: user regex... for matching the conditions for name.

File: lib/registration/customer.ex [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+      def check_aadhar(aadhar) do
+        if is_integer(aadhar) == true do
+          IO.inspect(aadhar)
+         count =
+          aadhar
+          |> Integer.digits
+          |> length
+
+          if count == 12 do
```
Comment (blackode) [inline lib/registration/customer.ex]: use regex... for matching the conditions for name.

**Review by blackode — CHANGES_REQUESTED** [review]

**Review by blackode — APPROVED** [review]
Remove inspect debug lines and rest is LGTM

File: lib/registration/customer.ex [hunk 9]
```diff
@@ -28,11 +28,13 @@ defmodule Registration.Customer do
     customer_data = %{name: name, aadhar: aadhar}
    if Registration.Customer.is_valid(customer_data) do
 
-       "your KYC is successfully updated."
+     IO.puts  "your KYC is successfully updated."
      else
-       "Please check the details you have entered and try again!"
+      IO.puts "Please check the details you have entered and try again!\n\n"
+       Registration.Customer.get_data_from_customer
```
Comment (blackode) [inline lib/registration/customer.ex]:
```suggestion
       Registration.Customer.get_data_from_customer()
```

**Review by blackode — CHANGES_REQUESTED** [review]

File: lib/registration/customer.ex (hunk 8 above)
Comment (singlefleck) [reply lib/registration/customer.ex]: used regex

Comment (singlefleck) [reply lib/registration/customer.ex]: image

File: test/customer_test.exs (hunk 2 above)
Comment (singlefleck) [reply test/customer_test.exs]: removed

**Review by blackode — APPROVED** [review]

**Pull request merged**